### DISCUSSION TITLE
How to write to a new pdf ?

### DISCUSSION THREAD

**Discussion by Laxmi530** [py-pdf/pypdf#1826] [Q&A]
Hi,
I have a multi-page pdf first extract the text from pdf after that doing some manipulation in text and then want to write the text to a new pdf file. I didn't find any guide in the documentation so from the code I saw the method `PdfWriter` and `add_page` but not getting how to do.

Can someone please guide me how to do this.
Thanking you in advance.

**Comment by pubpub-zz** [comment]
although pypdf is a library that is designed to manipulating(modify/assemble/extract) pdf; creating a page with pypdf is more difficult : you will have to build from scratch the whole content.
I would morelikely recommend you the pdf printer available in windows, ubuntu (eg. cups-pdf),... or some conversion program (example not tested https://wkhtmltopdf.org/, PIL for images to PDF)

**Reply by MartinThoma** [reply]
In order to create new PDF documents, you can use https://pypi.org/project/fpdf2/

**Reply by Laxmi530** [reply]
Thanks @pubpub-zz @MartinThoma for your response.
Do you have any idea wanted to mask certain fields in the pdf file, suppose like account number, phone number and name? 

Thanking you for your help.

**Reply by MartinThoma** [reply]
I'm happy to help :-) 

What I you mean by "mask"

**Reply by Laxmi530** [reply]
Hai @MartinThoma,
I mean PDF Redaction.
In the below pdf in 1st page wanted to replace `1234567890` with `123 456 7890` and in 2nd page also wanted to replace the `Mobile number`.
Can you please guide me the process.
[Test_Document.pdf](https://github.com/py-pdf/pypdf/files/11625970/Test_Document.pdf)

Thanking you in advance.

**Reply by pubpub-zz** [reply]
PDF is an electronic printed paper format : within a PDF, you can not presume how the text has been specified : the sequence 123456789 can be specified as a single sequence or two or three with some absolute or relative positioning between each bloc : substituion is something very tricky : the easiest solution I would propose would be to open your pdf document with libre office, which will convert it into a document, modify it and then reprint/export it back into a PDF
